skills_ml/algorithms/skill_extractors/fuzzy_match.py

```diff
@@ -50,50 +56,51 @@ def _skills_lookup(self) -> set:
             next(reader)
             index = 3
             generator = (self.reg_ex(row[index]) for row in reader)
-
             return set(generator)
 
-    def ngrams(self, sent, n):
-        sent_input = sent.split()
-        output = []
-        for i in range(len(sent_input)-n+1):
-            output.append(sent_input[i:i+n])
-        return output
-
-    def fuzzy_matches_in_sentence(self, skill, sentence):
-        N = len(skill.split())
-        doc = self.ngrams(sentence, N)
-        doc_join = [b" ".join(d) for d in doc]
-
-        for dj in doc_join:
-            ratio = fuzz.partial_ratio(skill, dj)
-            if ratio > self.match_threshold:
-                yield CandidateSkill(
-                    skill_name=skill,
-                    matched_skill=dj,
-                    confidence=ratio,
-                    context=sentence.decode('utf-8')
-                )
+    def ngrams(self, sent: Text, N: int) -> Generator[Text, None, None]:
+        """Yield ngrams from sentence
+
+        Args:
+            sent (string) A sentence
+            N (int) The maximum N-gram length to create. 
+
+        Yields: (string) n-grams in sentence from n=1 up to the given N
+        """
+        sent_input = self.nlp.word_tokenize(sent)
+        for n in range(1, N):
+            for i in range(len(sent_input)-n+1):
+                yield " ".join(sent_input[i:i+n]).lower()
+
+    @cachedproperty
+    def symspell(self):
+        """A SymSpell lookup based on the object's list of skills"""
+        ss = SymSpell(max_dictionary_edit_distance=4)
+        ss.create_dictionary(list(self.lookup))
+        return ss
 
     def candidate_skills(self, source_object: Dict) -> CandidateSkillYielder:
         document = self.transform_func(source_object)
         sentences = self.nlp.sentence_tokenize(document)
 
         for sent in sentences:
-            exact_matches = set()
-            for cs in self.candidate_skills_in_context(sent): 
-                yield cs
-                if cs.skill_name not in exact_matches:
-                    exact_matches.add(cs.skill_name)
-
-            for skill in self.lookup:
-                if skill in exact_matches:
-                    continue
-                ratio = fuzz.partial_ratio(skill, sent)
-                # You can adjust the partial of matching here:
-                # 100 => exact matching 0 => no matching
-                if ratio > self.match_threshold:
-                    logging.info('Found fuzzy matches passing threshold in %s', sent)
-                    for match in self.fuzzy_matches_in_sentence(skill, sent.encode('utf-8')):
-                        logging.info('Returning fuzzy match %s in sent: %s', match, sent)
-                        yield match
+            for phrase in self.ngrams(sent, 5):
```

Comment:
do we want to make the maximum N-gram length an argument instead of hard coded? and also other parameter like `max_distance`